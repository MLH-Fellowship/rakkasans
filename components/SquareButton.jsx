import React from 'react';
import {
  View, Text, StyleSheet, TouchableOpacity,
} from 'react-native';

import Colors from '../constants/Colors';
import Icon from './Icon';
import Dimensions from '../constants/Dimensions';

export default function SquareButton({
  text,
  onPress,
  color = Colors.blue5,
  name,
  buttonSize,
  textSize,
  iconSize = 50,
  hubIcon = false,
}) {
  const styles = StyleSheet.create({
    button: {
      backgroundColor: Colors.white,
      padding: 10,
    },
    container: {
      alignItems: 'center',
      justifyContent: 'center',
    },
    text: {
      textAlign: 'center',
      color: Colors.darkGray,
      paddingBottom: 6,
    },
  });
  return (
    <TouchableOpacity
      style={[styles.button, Dimensions.button]}
      onPress={onPress}
    >
      <View
        style={[styles.container, { height: buttonSize, width: buttonSize }]}
      >
        <Text style={[styles.text, Dimensions.font, { fontSize: textSize }]}>
          {text}
        </Text>
        <Icon name={name} iconColor={color} size={iconSize} hubIcon={hubIcon} />
      </View>
    </TouchableOpacity>
  );
}
