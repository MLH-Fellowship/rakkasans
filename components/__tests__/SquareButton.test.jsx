import React from 'react';
import renderer from 'react-test-renderer';

import SquareButton from '../SquareButton';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<SquareButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
