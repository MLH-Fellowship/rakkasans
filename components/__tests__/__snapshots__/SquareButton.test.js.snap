// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 15,
      "elevation": 5,
      "opacity": 1,
      "padding": 10,
      "shadowColor": "#a0a0a0",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.4,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333333",
            "paddingBottom": 6,
            "textAlign": "center",
          },
          Object {
            "fontFamily": "andale-mono",
          },
          Object {
            "fontSize": undefined,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
